src/main/java/com/hades/login/LoginInfoDAO.java

```diff
@@ -0,0 +1,49 @@
+package com.hades.login;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class LoginInfoDAO {
+
+	@PersistenceContext
+	private EntityManager manager;
+
+	public void save(LoginInfo user) {
+		manager.persist(user);
+	}
+
+	public Optional<LoginInfo> findBy(String login, String password) {
+		List<LoginInfo> resultList = manager
+				.createQuery("select u from " + LoginInfo.class.getSimpleName()
+						+ " u where u.login = :login and u.password = :password", LoginInfo.class)
+				.setParameter("login", login).setParameter("password", password).getResultList();
+		if (resultList.isEmpty()) {
+			return Optional.empty();
+		}
+		return Optional.ofNullable(resultList.get(0));
```

Comment:
Pode ser ``Optional.of(resultList.get(0))`` aqui pois temos certeza que temos um objeto aqui.